relay-server/src/services/processor.rs

```diff
@@ -1597,32 +1585,42 @@ impl EnvelopeProcessorService {
             })
         })?;
 
-        state.event_fully_normalized |= full_normalization;
+        event_fully_normalized |= full_normalization;
 
-        Ok(())
+        Ok(Some(event_fully_normalized))
     }
 
     /// Processes the general errors, and the items which require or create the events.
     fn process_errors(
         &self,
         state: &mut ProcessEnvelopeState<ErrorGroup>,
     ) -> Result<(), ProcessingError> {
+        let mut event_fully_normalized = state.envelope().event_fully_normalized();
+
         // Events can also contain user reports.
         report::process_user_reports(state);
 
         if_processing!(self.inner.config, {
             unreal::expand(state, &self.inner.config)?;
         });
 
-        event::extract(state, &self.inner.config)?;
+        event::extract(state, event_fully_normalized, &self.inner.config)?;
 
         if_processing!(self.inner.config, {
-            unreal::process(state)?;
-            attachment::create_placeholders(state);
+            if let Some(inner_event_fully_normalized) = unreal::process(state)? {
```

Comment:
Very ugly, but we have to have a tristate, since we need to represent the fact of not updating the event normalization boolean.

Reply:
This would be less awkward with an enum that encodes the tristate. Then you could do something like `event_fully_normalized.update(unreal::process(state)?);`